examples/platform/linux/CommissionableInit.cpp

```diff
@@ -99,6 +100,33 @@ CHIP_ERROR InitConfigurationManager(ConfigurationManagerImpl & configManager, Li
         configManager.StoreProductId(options.payload.productID);
     }
 
+    if (options.vendorName.HasValue())
+    {
+        const char * vendorName = options.vendorName.Value().c_str();
+        configManager.StoreVendorName(vendorName, strlen(vendorName));
+    }
+    if (options.productName.HasValue())
+    {
+        const char * productName = options.productName.Value().c_str();
+        configManager.StoreProductName(productName,strlen(productName));
+    }
+    if (options.hardwareVersionString.HasValue())
+    {
+        const char * hardwareVersionString = options.hardwareVersionString.Value().c_str();
+        configManager.StoreHardwareVersionString(hardwareVersionString,strlen(hardwareVersionString));
+    }
+    if (options.softwareVersionString.HasValue())
+    {
+        const char * softwareVersionString = options.softwareVersionString.Value().c_str();
+        configManager.StoreSoftwareVersionString(softwareVersionString,strlen(softwareVersionString));
```

Comment:
Right now, if you send in an invalid value, and this errors, it will fall back to the default silently. We've had problems with this before. It might be worthwhile to fail out here with a proper error message.

Reply:
fixed